src/plugin/file/kml/kmlsource.js

```diff
@@ -740,3 +741,72 @@ plugin.file.kml.KMLSource.prototype.restore = function(config) {
     this.setMinRefreshPeriod(config['minRefreshPeriod']);
   }
 };
+
+
+/**
+ * Creates a basic feature overlay used to animate features on the map.
+ * @protected
+ * @override
+ */
+plugin.file.kml.KMLSource.prototype.createAnimationOverlay = function() {
+  plugin.file.kml.KMLSource.superClass_.createAnimationOverlay.call(this);
+
+  // Hide all annotations
+  for (var i = 0, n = this.getFeatures().length; i < n; i++) {
```

Comment:
The runtime of `ol.source.Vector#getFeatures` is non-trivial because it creates a new array of features from the underlying r-tree. As a small optimization, call it once and save the result for the two places it's used in this function.